test/known_problems/should_fail/intersection_with_any_fail.erl

```diff
@@ -10,13 +9,6 @@
 any_refined_using_guard(X) when is_atom(X) ->
     X.
 
-%% X :: integer() & any() by the constraints
--spec intersection_using_constraits(X) when X :: integer(),
-                                            X :: any() ->
-                                                      atom().
-intersection_using_constraits(X) ->
-    X.
```

Comment:
If the syntax is corrected, i.e. `(X) -> atom() when X :: integer(), X :: any()`, (meaning X is some unknown subtype of integer) this should fail. Is this test case moved to another file? I don't see it.

Reply:
In a sense https://github.com/josefs/Gradualizer/pull/303/files#diff-101a1f8f433ecac6155ed3a0e816518e2ddf27a08f899bbe8785782cf91eca14 covers a similar case, but one more (#305 ) would not hurt

Reply:
They are different though. `5` is a subtype of `integer()` so we shouldn't report an error for this one (and if we do, we can call it a known problem). `atom()` isn't a subtype of `integer()` though, so that one is a type error whether `X :: any()` is present or not.

I'm sorry, it wasn't my intention to nag about these subtleties.

Reply:
Justice restored: #305